arroyo/processing/strategies/streaming/transform.py

```diff
@@ -239,9 +273,13 @@ def parallel_transform_worker_apply(
 
         try:
             result = function(message)
+        except InvalidMessages as e:
+            bad_messages += e.messages
+            i += 1
```

Comment:
The way this loop is written is a bit unintuitive to me. Is there any particular reason we can't just write:
```
i = 0
for message in input_batch:
    ...
```

Reply:
Function is called again with the index to start processing again from when `ValueTooLarge` is caught I believe.

https://github.com/getsentry/arroyo/blob/9b801a2892d6eb39367768482f102a38f5171de8/arroyo/processing/strategies/streaming/transform.py#L415-L437